### DISCUSSION TITLE
Telerecon Issues/Questions/Feedback

### DISCUSSION THREAD

**Discussion by sockysec** [sockysec/Telerecon#3] [General]
Feel free to use this forum to discuss, suggest features, or provide feedback on Telerecon.

**Comment by chengeniubi** [comment]
Pick a number: 1
Loading userdetails.py...
Enter target @username: @*****
Attempt 1 at connecting failed: TimeoutError:
Attempt 2 at connecting failed: TimeoutError:
Attempt 3 at connecting failed: TimeoutError:
Attempt 4 at connecting failed: TimeoutError:
Attempt 5 at connecting failed: TimeoutError:
Attempt 6 at connecting failed: TimeoutError:
Why, after I've installed and run it, does it show 'timeout'? I'm sure my network is smooth.

**Reply by sockysec** [reply]
Hi, thanks for using Telerecon.

A TimeoutError does likely indicate some connectivity issue, either with your internet or Telegram itself. I have been unable to replicate the issue myself, so it is likely a local issue.

To troubleshoot...

- Is your Telegram API credentials correct and set up in Telerecon correctly?
- When setting up Telerecon or trying to run the script, were you asked for a "Phone/Bot token"? And sent a verification code by/on Telegram to confirm the API connection/access? 
- Have you been able to use any of the other Telerecon modules successfully? (perhaps try census.py) 

Note: If Telegram is blocked in your country, this will likely prevent the desired connection. If your account or API access is somehow limited or blocked by Telegram, this will likely also prevent access (although, this is unlikely).

**Comment by yydeepfam** [comment]
After using it once to obtain user information, the tg account was banned.

**Reply by sockysec** [reply]
To clarify, you solely ran the user details script off of a @username and your tg account was subsequently banned? 

This issue is highly unusual and has never occurred during extensive testing (or been reported by any other users).
This script is within Telegram's T&C's so it should not result in this outcome.

- Was this a temporary rate limit or a full ban?
- Did you receive an error/issue message? If so what reason did TG give for the ban?
- Did you make any modifications to the script prior to running? (i.e. adjusting the REQUEST_DELAY)
- Was there something unique about the target of your query? (i.e. Perhaps Telegram closely monitoring API calls about that individual) 

The most likely scenario is that your account was banned by TG for some other reason or account behavior (i.e. breaching T&C's) and it just happened to coincide with the use of this script.

**Reply by yydeepfam** [reply]
Yes, I used the @username query and then I noticed that the telegram accounts on both PC and mobile were automatically logged out, and I logged in again to show that the account was banned, with no indication of other banning instructions. I wrote an unblock request email to tg customer service and haven't gotten a reply yet.
The +1 US number I'm using was just a pickup code to get it, I'm not sure how much reliability the account has, but I've been using the account normally for a couple months now and haven't done any other violations. I used the tool to make a query the account was immediately banned, so I'm sure that's the reason.
1/2. no official banning instructions were given, it only showed that my account was banned.
3. i'm a newbie, i just followed the installation steps and i don't think i changed anything.
4. the target I searched for was nothing special, normally I use @userinfobot to get the id of the target but there is no feedback so I wanted to try the tool .

**Reply by sockysec** [reply]
Thanks for the additional information.

This is certainly highly anomalous - unfortunately, I have been unable to replicate this issue myself, so I am unable to advise on why this may have occurred beyond the above reasons. There is no reason this should have occurred and to my knowledge, no other users have experienced this issue. So my best guess is that it was some local issue with your account, network, or behavior. 

Please let me know if TG customer service does provide additional elaboration as this may help narrow down your issue.

**Comment by AtariOverlord09** [comment]
Issue with Telerecon - Unexpected Session Reset

Dear Telerecon Creator,

I hope this message finds you well. I am writing to bring to your attention an issue I encountered while using Telerecon. First and foremost, I would like to express my appreciation for the tool and its functionalities.

However, I recently experienced an unexpected problem with Telerecon that I believe warrants attention. Following a session with the tool, an unforeseen error occurred, leading to the complete reset of all my sessions. Consequently, I lost access to my Telegram account, and the associated phone number.

I would be grateful if you could look into this matter and provide any insights or potential solutions. I am willing to provide additional information, including logs, if needed, to assist in troubleshooting this issue. Your prompt attention to this matter is highly appreciated.

Thank you for your dedication to Telerecon, and I look forward to your guidance on resolving this matter.

**Reply by sockysec** [reply]
Hi Atari, 

I'm glad you've been enjoying Telerecon, thanks for the positive feedback and logging this issue. Sorry to hear you have had an issue.

If possible, would you mind providing some additional information to inform issue testing?

- Which Telerecon module were you using when the error occurred? Or did this occur at some unspecified point after use?
- Was there anything anomalous about your use of Telerecon? 
- Was there a particular error message/pop-up? (Did TG give you a reason? Was it an account ban?)
- Did you make any modifications to the script before running it? (i.e. adjusting the REQUEST_DELAY)
- Is it possible that the issue could be local? i.e. to do with your IP/VPN/Network or the specific nature of your TG account's activity?
- Any other information that may be useful?

Cheers

**Comment by kadiroGMD** [comment]
hi , can you help me fix this error please, this accured when using option 2 and 3

Traceback (most recent call last):
  File "/home/kadiro/Telerecon/setup.py", line 68, in <module>
    exec(open("launcher.py").read())
  File "<string>", line 127, in <module>
  File "<string>", line 185, in <module>
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "<string>", line 145, in main
  File "<string>", line 30, in scrape_user_messages
  File "/home/kadiro/.local/lib/python3.9/site-packages/telethon/client/auth.py", line 657, in __aenter__
    return await self.start()
  File "/home/kadiro/.local/lib/python3.9/site-packages/telethon/client/auth.py", line 135, in _start
    await self.connect()
  File "/home/kadiro/.local/lib/python3.9/site-packages/telethon/client/telegrambaseclient.py", line 555, in connect
    self.session.auth_key = self._sender.auth_key
  File "/home/kadiro/.local/lib/python3.9/site-packages/telethon/sessions/sqlite.py", line 180, in auth_key
    self._update_session_table()
  File "/home/kadiro/.local/lib/python3.9/site-packages/telethon/sessions/sqlite.py", line 194, in _update_session_table
    c.execute('delete from sessions')
sqlite3.OperationalError: attempt to write a readonly database

**Comment by sockysec** [comment]
Hey, sorry for the slow reply.

From the traceback, this appears to be a permissions issue with Telethon's SQLite session/database, rather than Telerecon. I'd recommend double-checking that the Telethon package is installed properly in a non-read-only directory. You could try 'sudo pip install telethon' to ensure you install it with the correct permissions.  

Also, ensure that the user running the script has the correct write permissions. You could try running the relevant scripts with the sudo privilege to troubleshoot this.

Let me know if you have any further issues.

**Comment by israil0046** [comment]
in telerecon i choose option 3 (collets user information from a target channels) and the result is this 
![Screenshot (63)](https://github.com/user-attachments/assets/6fee5cac-01ba-4691-995a-ea5a9bb9c203)
 
what i do now?

**Reply by sockysec** [reply]
Apologies for the slow reply, I only just saw this.

The most likely cause is that some of the dependencies (i.e. numpy/pandas) are out of data. I suggest insuring that it is updated to the latest version.

Otherwise i'd suggest copy/pasting the error into ChatGpt and seeing if it can identify/troubleshoot the issue locally for you.

**Comment by klepaklepa** [comment]
Dear creator,
While using the Telerecon, I have faced with the problem, after entering the code, my telegram account has stolen. I have lost access to my account due to reset of all my sessions as it was tied to the virtual phone number. The chats were also deleted. 
I see that not only me had the similar problem. I don't know what exactly had happened, is this blocking of an account, just a reset or stealing by someone.
